internal/scheduler/scheduler.go

```diff
@@ -0,0 +1,181 @@
+// Package scheduler implements a simple scheduler that triggers the next item when its due time is
+// reached based on the list of upcoming items.
+package scheduler
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/kopia/kopia/internal/clock"
+	"github.com/kopia/kopia/internal/ctxutil"
+	"github.com/kopia/kopia/repo/logging"
+)
+
+var log = logging.Module("scheduler")
+
+// GetItemsFunc is a callback that returns items for the scheduler to consider.
+type GetItemsFunc func(ctx context.Context, now time.Time) []Item
+
+// Item is a describes item that can be scheduled with a function invoked the next time the item is due.
+type Item struct {
+	Description string
+	NextTime    time.Time
+	Trigger     func()
```

Comment:
nit: naming, the executed function is not really the trigger for the `Item`, it is the function to be executed when this item is due. I cannot think of a good name though ATM.

Are items going to be persisted at any point? Should JSON tags be preemptively added?

Reply:
that's not something that needs to be persistent.

Reply:
@jkowalski This was already merged with the scheduler PR.

What about the comment regarding the naming of the `Trigger func()`? Any ideas of a more appropriate name for it?

Reply:
I am also in two minds about this. The trigger() is actually being used like a trigger in maintenance scheduler instead of it being an interleaving operation.

I think that is the correct usage because otherwise you could end up blocking the scheduler. Let keep the name for now as it signals the desired usage.